### PR TITLE
Feature/erp order page search

### PR DESCRIPTION
**Pull request by julianzimmermann**

### CODE REVIEW

File: bundles/erp-order-page-search-rest-api/src/FondOfOryx/Glue/ErpOrderPageSearchRestApi/Model/Mapper/ErpOrderMapper.php [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+     * @param array $searchResults
+     *
+     * @return \Generated\Shared\Transfer\RestErpOrderPageSearchCollectionResponseTransfer
+     */
+    public function mapErpOrderResource(
+        array $searchResults
+    ): RestErpOrderPageSearchCollectionResponseTransfer {
+        $responseTransfer = (new RestErpOrderPageSearchCollectionResponseTransfer())->fromArray($searchResults, true);
+
+        foreach ($searchResults['erp-orders'] as $erpOrderData) {
```
Comment (julianzimmermann) [inline bundles/erp-order-page-search-rest-api/src/FondOfOryx/Glue/ErpOrderPageSearchRestApi/Model/Mapper/ErpOrderMapper.php:23]: move to or use const

File: bundles/erp-order-page-search-rest-api/src/FondOfOryx/Glue/ErpOrderPageSearchRestApi/Model/Mapper/ErpOrderMapper.php [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+            $erpOrder = new ErpOrderTransfer();
+            $erpOrder->fromArray($erpOrderData, true);
+            $responseTransfer->addErpOrder($this->mapErpOrderItemData($erpOrder, $erpOrderData['erp_order_items']));
```
Comment (julianzimmermann) [inline bundles/erp-order-page-search-rest-api/src/FondOfOryx/Glue/ErpOrderPageSearchRestApi/Model/Mapper/ErpOrderMapper.php:26]: move to or use const

File: bundles/erp-order-page-search/composer.json [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    "fond-of-oryx/erp-order": "dev-main",
+    "fond-of-oryx/erp-order-permission": "dev-main",
+    "spryker/customer": "^7.0.0",
+    "spryker/event-behavior": "^1.0.0",
+    "spryker/search": "^8.0.0",
+    "spryker/search-elasticsearch": "^1.0.0",
+    "spryker/synchronization-extension": "^1.0.0"
+  },
+  "require-dev": {
+    "fond-of-codeception/spryker": "dev-master",
```
Comment (julianzimmermann) [inline bundles/erp-order-page-search/composer.json:26]: version 1.0.0

File: bundles/erp-order-page-search/src/FondOfOryx/Client/ErpOrderPageSearch/Dependency/Client/ErpOrderPageSearchToSessionClientBridge.php [hunk 4]
```diff
@@ -0,0 +1,21 @@
+<?php
```
Comment (julianzimmermann) [inline bundles/erp-order-page-search/src/FondOfOryx/Client/ErpOrderPageSearch/Dependency/Client/ErpOrderPageSearchToSessionClientBridge.php:1]: could be removed i think

File: bundles/erp-order-page-search/src/FondOfOryx/Client/ErpOrderPageSearch/Dependency/Client/ErpOrderPageSearchToSessionClientInterface.php [hunk 5]
```diff
@@ -0,0 +1,7 @@
+<?php
```
Comment (julianzimmermann) [inline bundles/erp-order-page-search/src/FondOfOryx/Client/ErpOrderPageSearch/Dependency/Client/ErpOrderPageSearchToSessionClientInterface.php:1]: could be removed i think

File: bundles/erp-order-page-search/src/FondOfOryx/Client/ErpOrderPageSearch/ErpOrderPageSearchDependencyProvider.php [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+    }
+
+    /**
+     * @param \Spryker\Client\Kernel\Container $container
+     *
+     * @return \Spryker\Client\Kernel\Container
+     */
+    private function addSessionClient(Container $container): Container
+    {
+        $container[static::CLIENT_SESSION] = static function (Container $container) {
```
Comment (julianzimmermann) [inline bundles/erp-order-page-search/src/FondOfOryx/Client/ErpOrderPageSearch/ErpOrderPageSearchDependencyProvider.php:56]: is not used. could be removed when bridge an interface are removed

File: bundles/erp-order-page-search/src/FondOfOryx/Client/ErpOrderPageSearch/ErpOrderPageSearchFactory.php [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+
+        return $searchQuery;
+    }
+
+    /**
+     * @return \FondOfOryx\Client\ErpOrderPageSearch\Dependency\Client\ErpOrderPageSearchToSessionClientInterface
+     */
+    protected function getSessionClient(): ErpOrderPageSearchToSessionClientInterface
+    {
+        return $this->getProvidedDependency(ErpOrderPageSearchDependencyProvider::CLIENT_SESSION);
```
Comment (julianzimmermann) [inline bundles/erp-order-page-search/src/FondOfOryx/Client/ErpOrderPageSearch/ErpOrderPageSearchFactory.php:43]: could be removed

File: bundles/erp-order-page-search/src/FondOfOryx/Zed/ErpOrderPageSearch/Persistence/ErpOrderPageSearchQueryContainer.php [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+
+namespace FondOfOryx\Zed\ErpOrderPageSearch\Persistence;
+
+use Orm\Zed\ErpOrder\Persistence\ErpOrderQuery;
+use Spryker\Zed\Kernel\Persistence\AbstractQueryContainer;
+
+/**
+ * @method \FondOfOryx\Zed\ErpOrderPageSearch\Persistence\ErpOrderPageSearchPersistenceFactory getFactory()
+ */
+class ErpOrderPageSearchQueryContainer extends AbstractQueryContainer implements ErpOrderPageSearchQueryContainerInterface
```
Comment (julianzimmermann) [inline bundles/erp-order-page-search/src/FondOfOryx/Zed/ErpOrderPageSearch/Persistence/ErpOrderPageSearchQueryContainer.php:11]: maybe move to repository

File: bundles/erp-order-permission/composer.json [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+    {
+      "name": "Julian Zimmermann",
+      "email": "info@julianzimmermann.dev"
+    }
+  ],
+  "require": {
+    "php": ">=7.2"
+  },
+  "require-dev": {
+    "fond-of-codeception/spryker": "dev-master",
```
Comment (julianzimmermann) [inline bundles/erp-order-permission/composer.json:15]: 1.0

File: bundles/erp-order/composer.json [hunk 10]
```diff
@@ -12,8 +16,7 @@
     "php": ">=7.2",
     "fond-of-oryx/erp-order-extension": "dev-master",
     "fond-of-spryker/country": "dev-spryker_upgrade || dev-feature/paypal-refund",
-    "spryker/company-business-unit": "^1.0.0 || ^2.0.0",
-    "spryker/company-user": "^1.0.0 || ^2.0.0"
+    "spryker/company-business-unit": "^1.0.0 || ^2.0.0"
   },
   "require-dev": {
     "fond-of-codeception/spryker": "dev-master",
```
Comment (julianzimmermann) [inline bundles/erp-order/composer.json:22]: ^1.0

File: bundles/erp-order/src/FondOfOryx/Zed/ErpOrder/Communication/Console/ErpOrderConsole.php [hunk 11]
```diff
@@ -122,14 +122,14 @@ protected function getCreateErpOrderData(): ErpOrderTransfer
         $externalRef = md5((string)time());
         $ref = md5($externalRef);
```
Comment (julianzimmermann) [inline bundles/erp-order/src/FondOfOryx/Zed/ErpOrder/Communication/Console/ErpOrderConsole.php:123]: complete file was for testing purpose only, should be removed later :)

**Review by julianzimmermann — COMMENTED** [review]
see comments :)

File: bundles/erp-order-page-search-rest-api/composer.json [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+      "name": "József Geng",
+      "email": "gengjozsef86@gmail.com"
+    },
+    {
+      "name": "Julian Zimmermann",
+      "email": "info@julianzimmermann.dev"
+    }
+  ],
+  "require": {
+    "php": ">=7.2",
```
Comment (daniel-rose) [inline bundles/erp-order-page-search-rest-api/composer.json:16]: Use php >= 7.3 (7.2 EOL)

File: bundles/erp-order-page-search-rest-api/composer.json [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+    "fond-of-oryx/erp-order-page-search": "dev-main",
+    "spryker/glue-application": "^1.0.0",
+    "spryker/glue-application-extension": "^1.0.0"
+  },
+  "require-dev": {
+    "fond-of-codeception/spryker": "dev-master",
```
Comment (daniel-rose) [inline bundles/erp-order-page-search-rest-api/composer.json:22]: use fix version

File: bundles/erp-order-page-search/LICENSE [hunk 14]
```diff
@@ -0,0 +1,9 @@
+The MIT License (MIT)
+
+Copyright (c) 2020 FOND OF GmbH
```
Comment (daniel-rose) [inline bundles/erp-order-page-search/LICENSE:3]: 2021 please

File: bundles/erp-order-page-search/.gitignore [hunk 15]
```diff
@@ -0,0 +1,6 @@
+.idea
+
+.DS_Store
+
+vendor
+composer.lock
```
Comment (daniel-rose) [inline bundles/erp-order-page-search/.gitignore:6]: ignore src/Generated/ also

File: bundles/erp-order-page-search/README.md [hunk 16]
```diff
@@ -0,0 +1,54 @@
+# ERPOrderPageSearch
+[![Build Status](https://travis-ci.org/fond-of-oryx/erp-order-page-search.svg?branch=master)](https://travis-ci.org/fond-of-oryx/erp-order-page-search)
+[![PHP from Travis config](https://img.shields.io/travis/php-v/symfony/symfony.svg)](https://php.net/)
```
Comment (daniel-rose) [inline bundles/erp-order-page-search/README.md:3]: Remove travis lines...

File: bundles/erp-order-page-search/README.md [hunk 17]
```diff
@@ -0,0 +1,54 @@
...
+[![license](https://img.shields.io/github/license/mashape/apistatus.svg)](https://packagist.org/packages/fond-of-oryx/erp-order-page-search)
```
Comment (daniel-rose) [inline bundles/erp-order-page-search/README.md:4]: https://img.shields.io/github/license/mashape/apistatus.svg <-- wrong url.

File: bundles/erp-order-page-search/composer.json [hunk 18]
```diff
@@ -0,0 +1,42 @@
...
+      "name": "József Geng",
+      "email": "gengjozsef86@gmail.com"
+    },
+    {
+      "name": "Julian Zimmermann",
+      "email": "info@julianzimmermann.dev"
+    }
+  ],
+  "require": {
+    "php": ">=7.2",
```
Comment (daniel-rose) [inline bundles/erp-order-page-search/composer.json:16]: Look above. 7.2 EOL

File: bundles/erp-order-page-search/composer.json [hunk 19]
```diff
@@ -0,0 +1,42 @@
...
+  "prefer-stable": true,
+  "autoload": {
+    "psr-4": {
+      "FondOfOryx\\": "src/FondOfOryx/"
+    }
+  },
+  "autoload-dev": {
+    "psr-4": {
+      "FondOfOryx\\": "tests/FondOfOryx/",
+      "Generated\\": "tests/Generated/"
```
Comment (daniel-rose) [inline bundles/erp-order-page-search/composer.json:39]: that must be
`"Generated\\": "src/Generated/"`

File: bundles/erp-order-page-search/src/FondOfOryx/Client/ErpOrderPageSearch/Plugin/SearchExtension/AbstractErpOrderPageSearchQueryExpanderPlugin.php [hunk 20]
```diff
@@ -0,0 +1,83 @@
...
+    {
+        $filterCollection = [];
+        if (array_key_exists($resource, $filterData)) {
+            foreach ($filterData[$resource] as $filter) {
+                $filterTransfer = (new ErpOrderPageSearchFilterTransfer())->fromArray($filter, true);
+                if ($filterTransfer->getResource() === $resource && $filterTransfer->getField() === $field) {
+                    $filterCollection[$filterTransfer->getValue()] = $filterTransfer;
+                }
+            }
+        }
```
Comment (daniel-rose) [inline bundles/erp-order-page-search/src/FondOfOryx/Client/ErpOrderPageSearch/Plugin/SearchExtension/AbstractErpOrderPageSearchQueryExpanderPlugin.php:36]: Early exit?

File: bundles/erp-order-page-search/src/FondOfOryx/Client/ErpOrderPageSearch/Plugin/SearchExtension/ErpOrderPageSearchCompanyBusinessUnitUuidFilterQueryExpanderPlugin.php [hunk 21]
```diff
@@ -0,0 +1,113 @@
...
+                $uuids[] = $this->getCustomer()->getCompanyUserTransfer()->getCompanyBusinessUnit()->getUuid();
+            }
+
+            $searchQuery = $this->addTerms(
+                ErpOrderIndexMap::COMPANY_BUSINESS_UNIT_UUID,
+                $uuids,
+                $searchQuery
+            );
+
+            unset($this->currentUser);
```
Comment (daniel-rose) [inline bundles/erp-order-page-search/src/FondOfOryx/Client/ErpOrderPageSearch/Plugin/SearchExtension/ErpOrderPageSearchCompanyBusinessUnitUuidFilterQueryExpanderPlugin.php:78]: Early exit?

File: bundles/erp-order-page-search/src/FondOfOryx/Client/ErpOrderPageSearch/Plugin/SearchExtension/ErpOrderPageSearchOwnParamFilterQueryExpanderPlugin.php [hunk 22]
```diff
@@ -0,0 +1,107 @@
...
+                ErpOrderIndexMap::COMPANY_BUSINESS_UNIT_UUID,
+                array_values($companyBusinessUnitUuids),
+                $searchQuery
+            );
+            $searchQuery = $this->addTerms(
+                ErpOrderIndexMap::EXTERNAL_REFERENCE,
+                array_values($externalReferences),
+                $searchQuery
+            );
+            unset($this->currentUser);
```
Comment (daniel-rose) [inline bundles/erp-order-page-search/src/FondOfOryx/Client/ErpOrderPageSearch/Plugin/SearchExtension/ErpOrderPageSearchOwnParamFilterQueryExpanderPlugin.php:83]: Early exit?

File: bundles/erp-order-permission/LICENSE [hunk 23]
```diff
@@ -0,0 +1,9 @@
+The MIT License (MIT)
+
+Copyright (c) 2020 FOND OF GmbH
```
Comment (daniel-rose) [inline bundles/erp-order-permission/LICENSE:3]: 2021

File: bundles/erp-order-permission/README.md [hunk 24]
```diff
@@ -0,0 +1,39 @@
+# ERP Order
+[![Build Status](https://travis-ci.org/fond-of-oryx/erp-order-permission.svg?branch=master)](https://travis-ci.org/fond-of-oryx/erp-order-permission)
+[![PHP from Travis config](https://img.shields.io/travis/php-v/symfony/symfony.svg)](https://php.net/)
```
Comment (daniel-rose) [inline bundles/erp-order-permission/README.md:3]: Remove travis stuff...

File: bundles/erp-order-permission/README.md [hunk 25]
```diff
@@ -0,0 +1,39 @@
...
+[![license](https://img.shields.io/github/license/mashape/apistatus.svg)](https://packagist.org/packages/fond-of-oryx/erp-order-permission)
```
Comment (daniel-rose) [inline bundles/erp-order-permission/README.md:4]: https://img.shields.io/github/license/mashape/apistatus.svg --> wrong url

File: bundles/erp-order-permission/composer.json [hunk 26]
```diff
@@ -0,0 +1,31 @@
...
+  "description": "ERP Order Permission module",
+  "license": "MIT",
+  "authors": [
+    {
+      "name": "Julian Zimmermann",
+      "email": "info@julianzimmermann.dev"
+    }
+  ],
+  "require": {
+    "php": ">=7.2"
```
Comment (daniel-rose) [inline bundles/erp-order-permission/composer.json:12]: EOL. Please use >= 7.3

File: bundles/erp-order-permission/composer.json [hunk 27]
```diff
@@ -0,0 +1,31 @@
...
+  "prefer-stable": true,
+  "autoload": {
+    "psr-4": {
+      "FondOfOryx\\": "src/FondOfOryx/"
+    }
+  },
+  "autoload-dev": {
+    "psr-4": {
+      "FondOfOryx\\": "tests/FondOfOryx/",
+      "Generated\\": "tests/Generated/"
```
Comment (daniel-rose) [inline bundles/erp-order-permission/composer.json:28]: `"Generated\\": "src/Generated/"`

File: bundles/erp-order/composer.json [hunk 28]
```diff
@@ -12,8 +16,7 @@
     "php": ">=7.2",
```
Comment (daniel-rose) [inline bundles/erp-order/composer.json:16]: EOL. Please use >=7.3

File: composer.json [hunk 29]
```diff
@@ -6,10 +6,15 @@
...
       "name": "Daniel Rose",
       "email": "daniel-rose@gmx.de"
+    },
+    {
+      "name": "József Geng",
+      "email": "gengjozsef86@gmail.com"
     }
   ],
   "require": {
     "php": ">=7.2",
```
Comment (daniel-rose) [inline composer.json:16]: EOL. Please use >=7.3

**Review by daniel-rose — CHANGES_REQUESTED** [review]

File: dandelion.json [hunk 30]
```diff
@@ -1,46 +1,57 @@
...
         },
         "customer-statistics-rest-api": {
-            "url": "https://<GITHUB_TOKEN>@github.com/fond-of-oryx/customer-statistics-rest-api.git",
             "path": "bundles/customer-statistics-rest-api/",
             "version": "1.0.0"
         },
         "erp-order": {
-            "url": "https://<GITHUB_TOKEN>@github.com/fond-of-oryx/erp-order.git",
             "path": "bundles/erp-order/",
             "version": "alpha-0.0.1"
```
Comment (sourcecube) [inline dandelion.json:13]: incorrect version.. should be something like `1.0.0-alpha.0`

**Review by sourcecube — APPROVED** [review]

File: bundles/erp-order-page-search-rest-api/LICENSE [hunk 31]
```diff
@@ -0,0 +1,9 @@
+The MIT License (MIT)
+
+Copyright (c) 2020 FOND OF GmbH
```
[the remark this answers is not in the document]
Comment (gengjozsef) [reply bundles/erp-order-page-search-rest-api/LICENSE:3]: @daniel-rose fixed

File: bundles/erp-order-page-search-rest-api/composer.json (hunk 12 above)
Comment (gengjozsef) [reply bundles/erp-order-page-search-rest-api/composer.json]: @daniel-rose fixed

File: bundles/erp-order-page-search-rest-api/composer.json (hunk 13 above)
Comment (gengjozsef) [reply bundles/erp-order-page-search-rest-api/composer.json]: @daniel-rose fixed

File: bundles/erp-order-page-search-rest-api/src/FondOfOryx/Glue/ErpOrderPageSearchRestApi/Model/Mapper/ErpOrderMapper.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (gengjozsef) [reply bundles/erp-order-page-search-rest-api/src/FondOfOryx/Glue/ErpOrderPageSearchRestApi/Model/Mapper/ErpOrderMapper.php]: @daniel-rose i build in the constants

File: bundles/erp-order-page-search-rest-api/src/FondOfOryx/Glue/ErpOrderPageSearchRestApi/Model/Mapper/ErpOrderMapper.php [hunk 32]
```diff
@@ -0,0 +1,52 @@
...
+     * @return \Generated\Shared\Transfer\RestErpOrderPageSearchCollectionResponseTransfer
+     */
+    public function mapErpOrderResource(
+        array $searchResults
+    ): RestErpOrderPageSearchCollectionResponseTransfer {
+        $responseTransfer = (new RestErpOrderPageSearchCollectionResponseTransfer())->fromArray($searchResults, true);
+
+        foreach ($searchResults[self::SEARCH_RESULT_KEY_ERP_ORDERS] as $erpOrderData) {
+            $erpOrder = new ErpOrderTransfer();
+            $erpOrder->fromArray($erpOrderData, true);
```
Comment (daniel-rose) [inline bundles/erp-order-page-search-rest-api/src/FondOfOryx/Glue/ErpOrderPageSearchRestApi/Model/Mapper/ErpOrderMapper.php:28]: Maybe
```php
$erpOrder = (new ErpOrderTransfer())
    ->fromArray($erpOrderData, true);
```

Comment (daniel-rose) [reply bundles/erp-order-page-search-rest-api/src/FondOfOryx/Glue/ErpOrderPageSearchRestApi/Model/Mapper/ErpOrderMapper.php:28]: Maybe erpOrderTransfer will be a better name

**Review by daniel-rose — APPROVED** [review]

**Pull request merged**